### DISCUSSION TITLE
Efficient way to implement 'Load More' functionality?

### DISCUSSION THREAD

**Discussion by seb-jones** [livewire/livewire#3447] [1. Help] [answered]
All the resources I can find when googling around, such as [this one](https://www.nicesnippets.com/blog/laravel-livewire-load-more-example), seem to suggest storing the items per page in a public property, increasing that when a user clicks the 'Load More' button, and loading all the models in the `render` method. Surely this is not an efficient way to do 'Load More', since each time the user clicks the button it will reload *all* the models that are visible on the page - meaning the more the user clicks that button the bigger and slower things get. 

Ideally we would need to load **only** the new models but still leave the markup rendered by the previous pages. Is this something Livewire could do or would it make more sense to use AlpineJS for this, and just call a method using `$wire` to load the new items?

**Comment by joshhanley** [comment]
@seb-jones you could probably make use of nested components to do this efficiently, as once they are loaded in the browser, they run independently of each other.

So you could have a main (parent) component that contains a count and a load more button, which then creates the required number of child components, each of which would be responsible for a chuck of results, which would receive from that parent maybe it's starting offset and limit of results.

That way each time you click "Load more" that parent would add another child component to the end of the list, and the child component would only fetch say the "10" records it should show.

Hope this helps!

**Reply by seb-jones** [reply]
Interesting! I was not aware that nested components worked like that - the [docs](https://laravel-livewire.com/docs/2.x/nesting-components) don't mention that from what I can see?

Anyways I will give it a try and report back. Thanks very much :)

**Reply by joshhanley** [reply]
@seb-jones yeah it could be made clearer, but it's this part

> Nested components CAN accept data parameters from their parents, HOWEVER they are not reactive like props from a Vue component.

The reason they're not reactive is because they run standalone. It's why if you update a property in the parent, it's not reflected in the child, hence needing events to communicate between them.

**Comment by bilogic** [comment, marked as answer]
@seb-jones

http://www.ascsoftwares.com/technology/livewire-component-for-load-more-records/

I used this and it worked ok.

**Reply by seb-jones** [reply]
Right yeah, it looks like that tutorial is suggesting the same approach as joshhanley mentioned above. Thanks for sharing.

Out of curiosity, is your component dealing with a large amount of data? If so, does it perform more-or-less the same as the amount of data on the page increases?

**Reply by bilogic** [reply]
Hmm where is your area of concern?

I have about 3k records, currently loading 6 at a time as a test. I plan to load 50 when in production.
Database load largely depends on design, if done properly, Laravel handles it efficiently.
I don't think the DOM updates would be a concern here although my HTML is kept simple.

**Reply by bilogic** [reply]
@seb-jones pardon me, but would you happen to know anything about using view composers with livewire components? https://github.com/livewire/livewire/discussions/3453?

**Reply by seb-jones** [reply]
Sorry, I'm afraid not.
